[Performance] Performance REST API migration to GraphQL

bariqhibat opened the pull request:
**Description**

**Notes for Reviewers**
GraphQL performance subscription. When a new result comes, it'll publish the result to the channels, then the listeners would receive the data.

To reproduce:
1. Run the below query on GraphQL playground
Query
```graph
subscription{
  subscribePerfProfile(profileID: "44693737-3c17-485e-bdd2-c2f9be9c5dda"){
    meshery_id
    name
    mesh
  }
}
```
2. Go to [Performance profiles](http://localhost:9081/performance/profiles)
3. Run performance test for this specific profile (since the corresponding ID on the query above is for this profile)
![image](https://user-images.githubusercontent.com/54849895/130400332-186280b1-39eb-4de8-a521-6dea8773014c.png)
4. Once test finishes, you'll see a new performance result on the GraphQL playground
![image](https://user-images.githubusercontent.com/54849895/130400446-77b804c1-2dd8-4723-8134-e48925f19b69.png)



**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 

<!--
Thank you for contributing to Meshery! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

leecalcote commented:
👀

sudo-NithishKarthik commented on ui/components/MesheryPerformance/Dashboard.js:
```diff
@@ -43,18 +44,28 @@ function Dashboard({
+        // @ts-ignore
+        let result = res?.getPerformanceProfiles;
+        if (typeof result !== "undefined") {
+          updateProgress({ showProgress : false });
```
This can be moved outside of the if block ig, because, if the `result` is undefined, then the progress bar will not stop, which can be frustrating at times 😅

sudo-NithishKarthik commented on ui/components/MesheryPerformance/PerformanceProfiles.js:
```diff
@@ -86,26 +87,30 @@ function PerformanceProfile({ updateProgress, enqueueSnackbar, closeSnackbar })
+        // @ts-ignore
+        let result = res?.getPerformanceProfiles;
+        if (typeof result !== "undefined") {
+          updateProgress({ showProgress : false });
```
same here

bariqhibat commented:
Hey @sudo-NithishKarthik, thanks for the feedback. 

There's one more step to this actually, it is to update the cache store in our Relay. I've looked into it for a while but it's not really as intuitive as I thought, could you hop in to help me out with this?

The idea is to update by appending the newly-came data from subscription to the original query.

sudo-NithishKarthik commented:
Hey @bariqhibat , 

To answer your question, Seems like Relay [handles that automatically](https://relay.dev/docs/guided-tour/updating-data/graphql-subscriptions/), we just have to make sure we update the data in the store, for which we can use the methods that [they provide](https://relay.dev/docs/api-reference/store/#recordsourceselectorproxy) to work with Relay store, but, yes, I don't find it very intuitive as well 😕 .

And it makes sense because Relay basically wants to have full control over the store.

But, with our current implementation, we are not actually making use of relay's cache, yes, we are using `fetchQuery` which internally stores the data in Relay's store, but we are updating the React local state whenever we get a payload from subscriptions, this way, we do not have to fight with relay for control over the store 😝 .

And, this is only till we setup Redux, after that, we will only have to update the global store.

dhruv0000 commented on internal/graphql/resolver/addons.go:
```diff
@@ -21,7 +22,7 @@ func (r *Resolver) getAvailableAddons(ctx context.Context, provider models.Provi
 	}
 
-	addonlist, err := model.GetAddonsState(selectors, r.Config.KubeClient, provider)
+	addonlist, err := services.GetAddonsState(selectors, r.Config.KubeClient, provider)
```
Why chance from `model` to `service` package name? We have been using `model` convention in Meshery.

bariqhibat replied:
Since I need to decouple "model"-definition generated by gqlgen and the "models"-function like

dhruv0000 replied:
Why the need to decouple? The resolvers are both generated and hand written, and we still use the same package for it.

bariqhibat replied:
For me, it doesn't make sense to put the type definitions for the resolvers to be in the same package as functions for talking to the database. The other thing is that it's difficult to connect the packages due to cyclic imports.

bariqhibat commented:
Putting this to draft since need to wait for state management for the subscription update

bariqhibat commented:
Migrated all REST API to GraphQL resolvers on `/performance`

![image](https://user-images.githubusercontent.com/54849895/135655141-a523b3ed-5b48-429b-88b5-10eb72f77db7.png)

bariqhibat commented:
Updated the sheet on 
https://docs.google.com/spreadsheets/d/1ABJCbfQRi0uN_YoP2kmHZ-lTI4S4QuvRk1_unomKNRE/edit#gid=1241442399

Added new Get Resolver for fetchAllResults, getPerformanceProfiles

leecalcote commented:
> Updated the sheet on https://docs.google.com/spreadsheets/d/1ABJCbfQRi0uN_YoP2kmHZ-lTI4S4QuvRk1_unomKNRE/edit#gid=1241442399
> 
> Added new Get Resolver for `fetchAllResults`, `getPerformanceProfiles`

Thank you, @bariqhibat. Do you consider this PR ready for merge?

leecalcote commented on handlers/load_test_handler.go:
```diff
@@ -456,6 +458,27 @@ func (h *Handler) executeLoadTest(ctx context.Context, req *http.Request, testNa
+	// 	TestID:             &result.TestID,
+	// 	RunnerResults:      result.Result,
+	// 	ServerMetrics:      &serverMetrics,
+	// 	ServerBoardConfig:  &serverBoardConfig,
```
@dhruv0000 ServerBoard synonymous with StaticBoard? // @nehaLohia27

bariqhibat commented:
> Thank you, @bariqhibat. Do you consider this PR ready for merge?

Yes, I've tested this manually and it's working fine.

leecalcote commented:
Thank you, @bariqhibat. Moving forward...
